android/java/org/chromium/chrome/browser/preferences/website/BraveShieldsContentSettings.java

```diff
@@ -0,0 +1,181 @@
+            websitePreferenceBridge.getContentSettingsExceptions(ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT) :
+            websitePreferenceBridge.getContentSettingsExceptions(ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT);
+
+        for (ContentSettingException exception : exceptions) {
+            String pattern = exception.getPattern();
+
+            pattern = CutWwwPrefix(pattern);
+            if (!pattern.equals(host)) {
+                continue;
+            }
+
+            if (ContentSettingValues.ALLOW == exception.getContentSetting()) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        if (incognitoTab) {
+          //for incognito tab inherit settings from normal tab
+          return isJavaScriptEnabled(false, host);
+        }
+
+        if (!PrefServiceBridge.getInstance().isCategoryEnabled(ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static void setJavaScriptBlock(boolean incognitoTab, String host, boolean block, 
+            boolean fromTopShields, String resourceIndentifier) {
+        String hostForAccessMap = CutWwwPrefix(host);
+
+        int setting = ContentSettingValues.ALLOW;
+        if (block && !fromTopShields) {
+            setting = ContentSettingValues.BLOCK;
+        }
+
+        if (fromTopShields) {
+          // when change comes from Top Shields:
+          // in anyway do not touch hostSettings string
+          // block is false => we should allow js
+          // block is true => we should switch js back according to hostSettings string
+          if (!block) {
+            setting = ContentSettingValues.ALLOW;
+          } else {
+            setting = (!BraveShieldsContentSettingsJni.get().getShields(incognitoTab, host, resourceIndentifier).equals(allowResource)) ? 
+                ContentSettingValues.ALLOW : ContentSettingValues.BLOCK;
+          }
+        }
+
+        if (incognitoTab) {
+            // TODO JavaScript for incognito profiles. Check that
+            // https://github.com/brave/browser-android-tabs/commit/e1dd6f7797398155d640303e05241f9fb2b433f9#diff-e1d5c8c446116e371020d44baa09d09bR189
+            // PrefServiceBridge.getInstance().nativeSetContentSettingForPatternIncognito(
+            //           ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT, host,
+            //           setting);
+            PrefServiceBridge.getInstance().nativeSetContentSettingForPattern(
+                        ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT, host,
+                        setting);
+        } else {
+            PrefServiceBridge.getInstance().nativeSetContentSettingForPattern(
+                      ContentSettingsType.CONTENT_SETTINGS_TYPE_JAVASCRIPT, host,
+                      setting);
+        }
+    }
+
+    private static String CutWwwPrefix(String host) {
```

Comment:
this is not the equivalent of Origin. I think we should pass the full urls to the brave_shields helper methods and let them handle origin